packages/radio-group/src/Radio.tsx

```diff
@@ -1,76 +1,216 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { colors } from '@leafygreen-ui/theme';
+import { uiColors } from '@leafygreen-ui/palette';
+import { createDataProp } from '@leafygreen-ui/lib';
 import { css, cx } from '@leafygreen-ui/emotion';
-import Variant from './Variant';
+import { RadioGroupProps } from './RadioGroup';
+import { Variant, Size } from './types';
+
+const styledDiv = createDataProp('styled-div');
+
+const containerMargin = css`
+  margin-top: 8px;
+`;
+
+const smallOffset = css`
+  margin-top: -2px;
+`;
+
+const defaultOffset = css`
+  margin-top: 2px;
+`;
+
+const labelVariantStyle = {
+  [Variant.Default]: {
+    base: css`
+      color: ${uiColors.gray.dark2};
+    `,
+
+    disabled: css`
+      color ${uiColors.gray.base};
+    `,
+  },
+
+  [Variant.Light]: {
+    base: css`
+      color: ${uiColors.white};
+    `,
+
+    disabled: css`
+      color: ${uiColors.gray.light1};
+    `,
+  },
+};
 
 const labelStyle = css`
-  cursor: pointer;
-  margin-bottom: 5px;
-  display: block;
+  display: flex;
+  align-items: flex-start;
+  font-size: 14px;
+  line-height: 20px;
 `;
 
+// Note colors are not in our palette
+const inputVariantStyle = {
+  [Variant.Default]: css`
+    &:checked + ${styledDiv.selector} {
+      background-color: #2798bd;
+      border-color: #2798bd;
+
+      &:after {
+        transform: scale(1);
+      }
+    }
+
+    &:hover + ${styledDiv.selector}:before {
+      border: 3px solid rgba(184, 196, 194, 0.3);
+    }
+
+    &:focus + ${styledDiv.selector}:before {
+      border: 3px solid #9dd0e7;
+    }
+
+    &:disabled + ${styledDiv.selector} {
+      border-color: ${uiColors.gray.light1};
+      background-color: ${uiColors.gray.light2};
+    }
+  `,
+
+  [Variant.Light]: css`
+    &:checked + ${styledDiv.selector} {
+      background-color: #43b1e5;
+      border-color: #43b1e5;
+
+      &:after {
+        transform: scale(1);
+      }
+    }
+
+    &:hover + ${styledDiv.selector}:before {
+      border: 3px solid rgba(255, 255, 255, 0.2);
```

Comment:
Using a border for the hover and focus states prevents us from transitioning the state like we do in other components. We should use the same method for this border as we do in components like RadioBoxGroup.